components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -908,7 +908,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                 throw err;
             }
 
-            let projectEnvVarsPromise = this.internalGetPublicProjectEnvVars(workspace.projectId);
+            let projectEnvVarsPromise = this.internalGetProjectEnvVars(workspace.projectId);
```

Comment:
With the filtering moved to another level, I believe you can delete the `internalGet[...]ProjectEnvVars` helper function, and just call this here:

```suggestion
            let projectEnvVarsPromise = this.projectsService.getProjectEnvironmentVariables(workspace.projectId);
```

Reply:
It's still helpful to handle the "outside of a project" case